src/dataframe/dataframe.jl

```diff
@@ -119,7 +119,7 @@ mutable struct DataFrame <: AbstractDataFrame
                 throw(DimensionMismatch("columns must be 1-dimensional"))
             end
         end
-        new(columns, colindex)
+        new(Vector{AbstractVector}(columns), colindex)
```

Comment:
Would `convert` be OK here or would aliasing be a problem? Else, this change may have a performance impact when creating lots of data frames, like in `by`.

Same elsewhere.

Reply:
also see a comment below